core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -788,6 +788,128 @@ class GroupCoordinatorTest extends JUnitSuite {
     assertNotEquals(firstGenerationId, secondJoinResult.generationId)
   }
 
+  @Test
+  def testSecondMemberPartiallyJoinAndTimeout() {
```

Comment:
A brief comment describing the test scenario would be helpful since these tests are complicated.

Reply:
comment added above.